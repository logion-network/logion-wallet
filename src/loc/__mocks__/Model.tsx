import {
    rejectProtectionRequest,
    acceptProtectionRequest,
} from './ModelMock';

export {
    rejectProtectionRequest,
    acceptProtectionRequest,
};
