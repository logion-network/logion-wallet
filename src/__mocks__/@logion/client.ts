import { CheckResultType } from "@logion/client/dist/CollectionItem.js";
import { DefaultSignAndSendStrategy } from "@logion/client/dist/Signer.js";

import {
    LogionClient,
    AccountTokens,
    toIsoString,
    fromIsoString,
    ClosedCollectionLoc,
    DraftRequest,
    EditableRequest,
    OpenLoc,
    LocRequestState,
    LocsState,
    HashOrContent,
} from '../LogionClientMock';

export {
    LogionClient,
    AccountTokens,
    toIsoString,
    fromIsoString,
    CheckResultType,
    ClosedCollectionLoc,
    DraftRequest,
    EditableRequest,
    OpenLoc,
    DefaultSignAndSendStrategy,
    LocRequestState,
    LocsState,
    HashOrContent,
}

export { MimeType } from "@logion/client/dist/Mime.js";
