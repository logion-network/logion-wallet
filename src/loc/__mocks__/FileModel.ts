export { getAllDeliveries, getAllCollectionDeliveries } from './FileModelMock';
