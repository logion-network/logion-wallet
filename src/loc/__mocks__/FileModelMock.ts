export let getAllDeliveries = jest.fn().mockResolvedValue({});
