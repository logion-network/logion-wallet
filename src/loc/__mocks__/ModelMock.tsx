export let rejectRequest = jest.fn();

export let acceptRequest = jest.fn();

export function setAcceptRequest(mockFn: any) {
    acceptRequest = mockFn;
}

export let setAssetDescription = jest.fn();

export let rejectProtectionRequest = jest.fn();

export function setRejectProtectionRequest(mockFn: any) {
    rejectProtectionRequest = mockFn;
}

export let acceptProtectionRequest = jest.fn();

export function setAcceptProtectionRequest(mockFn: any) {
    acceptProtectionRequest = mockFn;
}

export function decision(legalOfficerAddress: string, decisions: any[]) {
    return decisions[0];
}

export let fetchRecoveryInfo = jest.fn();

export function setFetchRecoveryInfo(mockFn: any) {
    fetchRecoveryInfo = mockFn;
}

export let acceptLocRequest = jest.fn();

export function setAcceptLocRequest(mockFn: any) {
    acceptLocRequest = mockFn;
}

export let rejectLocRequest = jest.fn();

export function setRejectLocRequest(mockFn: any) {
    rejectLocRequest = mockFn;
}

export let addLink = jest.fn();
