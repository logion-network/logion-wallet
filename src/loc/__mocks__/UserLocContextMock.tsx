import { UUID } from "@logion/node-api/dist/UUID";
import { LocData } from "@logion/client/dist/Loc";

import { LocRequest } from "../../common/types/ModelTypes";
import { LocItem } from "../types";

let locState: any;

export function setLocState(value: any) {
    locState = value;
}

export let refresh = jest.fn();

let requestSof = jest.fn();

export function setRequestSof(fn: jest.Mock) {
    requestSof = fn;
}

let requestSofOnCollection = jest.fn();

export function setRequestSofOnCollection(fn: jest.Mock) {
    requestSofOnCollection = fn;
}

let locId: UUID;

export function setLocId(id: UUID) {
    locId = id;
}

let loc: LocData | null = null;

export function setLoc(request: LocData | null) {
    loc = request;
}

let locItems: LocItem[] = [];

export function setLocItems(items: LocItem[]) {
    locItems = items;
}

export let deleteMetadata = jest.fn();

export let deleteFile = jest.fn();

export let addFile = jest.fn().mockResolvedValue(undefined);

export let addMetadata = jest.fn().mockResolvedValue(undefined);

export function useUserLocContext() {

    return {
        refresh,
        locState,
        requestSof,
        requestSofOnCollection,
        locId,
        loc,
        deleteMetadata,
        deleteFile,
        locItems,
        addFile,
        addMetadata,
    }
}
