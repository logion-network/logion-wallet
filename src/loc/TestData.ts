import { LegalOfficerCase, UUID, Hash } from "@logion/node-api";
import { LocData, HashString } from "@logion/client";

export function buildLocRequest(locId: UUID, loc: LegalOfficerCase): LocData {
    return {
        ownerAddress: loc.owner,
        requesterAddress: loc.requesterAddress,
        requesterLocId: loc.requesterLocId,
        description: "Description",
        locType: loc.locType,
        createdOn: "",
        decisionOn: "",
        id: locId,
        status: loc.closed ? "CLOSED" : "OPEN",
        files: loc.files.map((locFile, index) => ({
            ...locFile,
            nature: HashString.fromValue(`File ${index}`),
            name: `File ${index}`,
            addedOn: "",
            published: false,
            restrictedDelivery: false,
            contentType: "text/plain",
            size: 42n,
            status: "DRAFT",
        })),
        metadata: loc.metadata.map((locFile, index) => ({
            ...locFile,
            name: HashString.fromValue(`Data ${index}`),
            value: HashString.fromValue(`Value ${index}`),
            addedOn: "",
            published: false,
            status: "DRAFT",
        })),
        links: loc.links.map((locFile, index) => ({
            ...locFile,
            name: `Link ${index}`,
            nature: HashString.fromValue(`Nature ${index}`),
            addedOn: "",
            target: locFile.id.toString(),
            published: false,
            status: "DRAFT",
        })),
        closed: false,
        verifiedIssuer: false,
        issuers: [],
    };
}
