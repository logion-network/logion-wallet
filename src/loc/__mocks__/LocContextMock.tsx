import { LocData, CollectionItem } from "@logion/client";
import { UUID, ValidAccountId } from "@logion/node-api";

import { LocItem } from "../LocItem";

let locState: any = {
    isLogionIdentity: () => false,
    isLogionData: () => true,
};

export function setLocState(value: any) {
    locState = value;
    if(value.data) {
        loc = {
            ...loc,
            ...locState.data(),
        }
    }
}

let locItems: LocItem[] = [];

export function setLocItems(items: LocItem[]) {
    locItems = items;
}

let locId = new UUID("aed4c6e4-979e-48ad-be6e-4bd39fb94762");

export function setLocId(id: UUID) {
    locId = id;
}

let loc: Partial<LocData> | null = {
    id: locId,
    locType: "Identity",
    requesterAccountId: ValidAccountId.polkadot("5Ew3MyB15VprZrjQVkpQFj8okmc9xLDSEdNhqMMS5cXsqxoW"),
    userIdentity: {
        firstName: "John",
        lastName: "Doe",
        email: "john.doe@logion.network",
        phoneNumber: "+1234",
    }
};

export function setLocRequest(request: Partial<LocData> | null) {
    loc = request;
}

export let refresh = jest.fn();

export let requestSof = jest.fn();

export function setRequestSof(fn: jest.Mock) {
    requestSof = fn;
}

export let deleteMetadata = jest.fn();

export let deleteLink = jest.fn();

let collectionItems: CollectionItem[] = [];

export function setCollectionItems(items: CollectionItem[]) {
    collectionItems = items;
}

async function mutateLocState(mutator: (state: any) => Promise<any>): Promise<void> {
    await mutator(locState);
}

export function useLocContext() {
    return {
        linkLoc: {

        },
        locId,
        loc,
        locItems,
        refresh,
        requestSof,
        deleteMetadata,
        deleteLink,
        checkResult: { result: "NONE" },
        locState,
        collectionItems,
        mutateLocState,
    };
}
