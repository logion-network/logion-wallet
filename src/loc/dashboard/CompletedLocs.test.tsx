import { shallowRender } from "src/tests";
import CompletedLocs from "./CompletedLocs";
import { LocRequestState, LocRequestStatus } from "@logion/client";
import { DateTime } from "luxon";
import { PATRICK } from "src/common/TestData";
import { setViewer } from "src/common/__mocks__/CommonContextMock";

jest.mock("src/common/CommonContext");

describe("CompletedLocs", () => {

    it("renders for user", () => {
        setViewer("User");
        const locs = [
            mockLoc({
                ownerAddress: PATRICK.address,
                closedOn: DateTime.fromISO("2024-03-19T16:00:00"),
                status: "CLOSED",
            }),
            mockLoc({
                ownerAddress: PATRICK.address,
                closedOn: DateTime.fromISO("2024-03-19T16:00:00"),
                status: "OPEN",
                voidInfo: {},
            }),
        ];
        const element = shallowRender(<CompletedLocs locDetailsPath={ () => "" } locs={ locs } />);
        expect(element).toMatchSnapshot();
    });

    it("renders for LLO", () => {
        setViewer("LegalOfficer");
        const locs = [
            mockLoc({
                userIdentity: {},
                closedOn: DateTime.fromISO("2024-03-19T16:00:00"),
                status: "CLOSED",
            }),
            mockLoc({
                userIdentity: {},
                closedOn: DateTime.fromISO("2024-03-19T16:00:00"),
                status: "OPEN",
                voidInfo: {},
            }),
        ];
        const element = shallowRender(<CompletedLocs locDetailsPath={ () => "" } locs={ locs } />);
        expect(element).toMatchSnapshot();
    });
});

function mockLoc(args: {
    ownerAddress?: string,
    userIdentity?: object,
    status: LocRequestStatus,
    closedOn: DateTime,
    voidInfo?: object,
}): LocRequestState {
    return {
        data: () => args,
    } as unknown as LocRequestState;
}
