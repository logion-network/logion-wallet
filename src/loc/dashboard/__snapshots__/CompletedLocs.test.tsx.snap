// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CompletedLocs renders for LLO 1`] = `
<Table
  columns={
    Array [
      Object {
        "align": "left",
        "header": "Requester",
        "render": [Function],
      },
      Object {
        "align": "left",
        "header": "Description",
        "render": [Function],
      },
      Object {
        "header": "Status",
        "render": [Function],
        "width": "140px",
      },
      Object {
        "align": "center",
        "header": "Closing date",
        "render": [Function],
        "width": "200px",
      },
      Object {
        "header": "Action",
        "render": [Function],
      },
    ]
  }
  data={
    Array [
      Object {
        "closedOn": "2024-03-19T16:00:00.000Z",
        "status": "CLOSED",
        "userIdentity": Object {},
      },
      Object {
        "closedOn": "2024-03-19T16:00:00.000Z",
        "status": "OPEN",
        "userIdentity": Object {},
        "voidInfo": Object {},
      },
    ]
  }
  renderEmpty={[Function]}
/>
`;

exports[`CompletedLocs renders for user 1`] = `
<Table
  columns={
    Array [
      Object {
        "align": "left",
        "header": "Legal officer",
        "render": [Function],
      },
      Object {
        "align": "left",
        "header": "Description",
        "render": [Function],
      },
      Object {
        "header": "Status",
        "render": [Function],
        "width": "140px",
      },
      Object {
        "align": "center",
        "header": "Closing date",
        "render": [Function],
        "width": "200px",
      },
      Object {
        "header": "Action",
        "render": [Function],
      },
    ]
  }
  data={
    Array [
      Object {
        "closedOn": "2024-03-19T16:00:00.000Z",
        "ownerAccountId": ValidAccountId {
          "anyAccountId": AnyAccountId {
            "address": "vQx5kESPn8dWyX4KxMCKqUyCaWUwtui1isX6PVNcZh2Ghjitr",
            "type": "Polkadot",
          },
        },
        "status": "CLOSED",
      },
      Object {
        "closedOn": "2024-03-19T16:00:00.000Z",
        "ownerAccountId": ValidAccountId {
          "anyAccountId": AnyAccountId {
            "address": "vQx5kESPn8dWyX4KxMCKqUyCaWUwtui1isX6PVNcZh2Ghjitr",
            "type": "Polkadot",
          },
        },
        "status": "OPEN",
        "voidInfo": Object {},
      },
    ]
  }
  renderEmpty={[Function]}
/>
`;
